sound/soc/sof/ipc4-topology.c

```diff
@@ -205,6 +214,35 @@ static int sof_ipc4_setup_process_init_byte_kcontrol(struct snd_sof_widget *swid
 		process->init_byte_kcontrol = control_data;
 	}
 
+	module_in_basefw = sof_ipc4_module_is_in_basefw(swidget->module_info);
+	if (!module_in_basefw) {
+		struct sof_ipc4_base_module_cfg_ext *base_ext;
+		int size;
+
+		/*
+		 * Each loadable module must has a byte kcontrol for its instance initialization,
+		 * we are safe without nullable check for control_data here.
+		 */
+		control_data = process->init_byte_kcontrol;
+		base_ext = (struct sof_ipc4_base_module_cfg_ext *)control_data->data->data;
+		if (base_ext->num_sink_pin_fmts < 1 || base_ext->num_source_pin_fmts < 1) {
```

Comment:
it would be better this way: if (!base_ext->num_sink_pin_fmts || !base_ext->num_source_pin_fmts)